packages/storybook/stories/components/file-upload/documentation.mdx

```diff
@@ -0,0 +1,122 @@
+import { Canvas, Meta, Markdown } from '@storybook/blocks';
+import SpecificationsFileUpload from '@ovhcloud/ods-components/src/components/file-upload/documentation/spec.md?raw';
+import { Banner } from '../../banner';
+import { DocNavigator } from '../../doc-navigator';
+import * as FileUploadStories from './file-upload.stories';
+import { LINK_ID } from '../../zeroheight';
+
+<Meta of={ FileUploadStories } name='Documentation' />
+
+<Banner of={ FileUploadStories } />
+
+# Overview
+
+File-upload allow users to select one or more files to upload to a specific location:
+
+<Canvas of={ FileUploadStories.Overview } sourceState='none' />
+
+<DocNavigator of={ FileUploadStories } linkId={ LINK_ID.FILE_UPLOAD } />
+
+<Markdown>{ SpecificationsFileUpload }</Markdown>
+
+# How to use?
+
+The component does not handle any actual upload, nor it knows the status of each file (regarding progress or error),
+as the upload process is different for every integration.
+
+All the files you'll receive from this component are typed as `OdsFile`.
+
+This is an extend of the native `File` interface with optional attribute that you can set to alter the display:
+- `error`: set this if the upload of the file fail, to display an error message near it on the component file list.
```

Comment:
set this if the upload of the file fails, in order to display an error message below the file list.*